app/src/main/java/org/mozilla/webmaker/web/javascript/WebAppInterface.java

```diff
@@ -140,7 +139,7 @@ public void setMemStorage(String key, final String value) {
     @JavascriptInterface
     public void setMemStorage(String key, final String value, final boolean global) {
         if (!global) key = key.concat(mPrefKey);
-        MemStorage.sharedStorage().put(key, value);
+        api.queue(key, value);
```

Comment:
the old MemStorage calls overwrite already cached data, which the new system doesn't do: it just queues "a next data payload", so we want to make sure to clear a queue before putting the data in, so that it's the only element:
```
api.clearPayloads(key);
api.queue(key, value);
```